localstack/services/cloudformation/models/iam.py

```diff
@@ -274,21 +274,22 @@ def _pre_delete(resource_id, resources, resource_type, func, stack_name):
         props = resource["Properties"]
         role_name = props["RoleName"]
 
-        # TODO: this should probably only remove the policies that are specified in the stack (verify with AWS)
-        # detach managed policies
-        for policy in iam_client.list_attached_role_policies(RoleName=role_name).get(
-            "AttachedPolicies", []
-        ):
-            iam_client.detach_role_policy(RoleName=role_name, PolicyArn=policy["PolicyArn"])
-        # delete inline policies
-        for inline_policy_name in iam_client.list_role_policies(RoleName=role_name).get(
-            "PolicyNames", []
-        ):
-            iam_client.delete_role_policy(RoleName=role_name, PolicyName=inline_policy_name)
-
-        # TODO: potentially remove this when stack resource deletion order is fixed (check AWS behavior first)
-        # cleanup instance profile
         try:
+            # TODO: this should probably only remove the policies that are specified in the stack (verify with AWS)
+            # detach managed policies
+            for policy in iam_client.list_attached_role_policies(RoleName=role_name).get(
+                "AttachedPolicies", []
+            ):
+                iam_client.detach_role_policy(RoleName=role_name, PolicyArn=policy["PolicyArn"])
```

Comment:
Might be better to wrap the `detach_...` and `delete_...` calls via `call_safe` instead. That way we don't miss the deletion of all other resources just because one of them failed to be found.

Reply:
Probably not a very likely scenario, but good to be on the safe side.. 👍 Updated the PR